### ISSUE TITLE
MFCRecorder.py connects then fails

### ISSUE THREAD

**Issue by dhearns** [beaston02/MFCRecorder#2]
Script cycles through a connections/disconnections as guest account and finally ends in the following error.  Followed readme to install requirements & complete setup.  Any Ideas?  Thanks in advance.

$ python3.5 MFCRecorder.py 
____________________Connection Status____________________
[2017-07-10 02:17:51,208] Connecting to MyFreeCams chat server xchat41...
Disconnected from MyFreeCams.  Reconnecting in 30 seconds...
Exception in callback _SelectorTransport._call_connection_lost(None)
handle: <Handle _SelectorTransport._call_connection_lost(None)>
Traceback (most recent call last):orded: []
  File "/usr/local/lib/python3.5/asyncio/events.py", line 125, in _run
    self._callback(*self._args)
  File "/usr/local/lib/python3.5/asyncio/selector_events.py", line 607, in _call_connection_lost
    self._protocol.connection_lost(exc)
  File "/usr/local/lib/python3.5/site-packages/mfcauto/client.py", line 25, in connection_lost
    self.client._disconnected()
  File "/usr/local/lib/python3.5/site-packages/mfcauto/client.py", line 195, in _disconnected
    self.loop.call_later(30, self.connect, self._logged_in)
  File "/usr/local/lib/python3.5/asyncio/base_events.py", line 418, in call_later
    timer = self.call_at(self.time() + delay, callback, *args)
  File "/usr/local/lib/python3.5/asyncio/base_events.py", line 430, in call_at
    raise TypeError("coroutines cannot be used with call_at()")
TypeError: coroutines cannot be used with call_at()

**Comment by sKanoodle** [comment]
It looks to me like an issue with mfcauto.py (linked in the readme). I don't really know enough about working async in python to tell you how good my fix is, but at least i got it working again. I changed line 195 in client.py from `self.loop.call_later(30, self.connect, self._logged_in)` to `self.loop.call_later(30, lambda: asyncio.async(self.connect(self._logged_in)))`.
This is just wrapping the async call in a lambda expression to make it work with call_at(). Maybe you can get better results when using a different call entirely, I don't know.

**Comment by dhearns** [comment]
Thanks @sKanoodle. Using your fix got the script working, it recorded for 9 minutes (~75 MB) , but eventually disconnected. Now when the script returns "Disconnected from MyFreeCams. reconnecting in 30 seconds..."  continuously.  When web browsing I'm greeted with "Disconnected from MyFreeCams, reconnecting"  Restarting the script and browser doesn't resolve the issue.  I wonder if my IP got blocked?  But the script does work.

**Comment by beaston02** [comment]
From what I can tell, MFC has made changes to not allow constant reconnection to their websocket which is used to get the list of online models. After so many new connections, it will block any more connections for sometime from the same IP address. 

I've been working on a solution for this (sorry for no updates). I can create a websocket connection and keep it open to continue to monitor all received messages, including models going into free chat to initiate a recording, but this method doesn't seem real efficient because not all status updates seem to be received, among other potential problems like a recording not starting, or ending early wouldn't be restarted until the model leaves the free chat and then goes back in. I would like to find a decent interval for creating a new connection and getting a "fresh" list of online models, but I have not yet found an interval that works efficiently. I just haven't came up with a decent solution yet. 

@skanoodle, thank you for posting the fix for mfcauto.

**Comment by dhearns** [comment]
@beaston02 you are correct.  I stopped the script and after about 15 minutes I was unblocked.  I restarted the script, was disconnected again but was able to confirm that the recording continued because the file size continued to grow.  I really appreciate all of your hard work & support on this! I'll keep an eye out for future updates.

@sKanoodle Thanks again for the fix.  I guess I can close this issue?

Man, you guys are geniuses. I don't know how you come up with this stuff!
